src/index.d.ts

```diff
@@ -14,7 +14,7 @@ declare namespace preact {
 		type: ComponentType<P> | string | null;
 		props: P & { children: ComponentChildren } | string | number | null;
 		key: Key;
-		ref: Ref<any> | null;
+		ref?: Ref<any> | null;
```

Comment:
I think that `?` is unnesseary if only `createVNode()` provides VNode.
https://github.com/preactjs/preact/blob/870cf60352c46a86e110dd598efe7be9179ba979/src/create-element.js#L53-L73

Reply:
Hmm, then I'll maybe have to dig deeper into this issue https://github.com/preactjs/preact/issues/1953

Reply:
Well I've re thought this through and if createVNode provides undefined as a ref then it should be ? since it's just undefined if the user doesn't provide one which makes it optional, no? Please correct me if I'm wrong

Reply:
> PS: do you agree with the other comments I made?

No, I do not.
Since I think `ref` or `ref?` is exsiting `ref` key or not in object.

```ts
let a : {b?: undefined} = {}
let c : {b: undefined} = {} // error
```

http://www.typescriptlang.org/play/index.html

Reply:
Yep, @38elements ist correct. The `?` sign means that the property doesn't exist which is different from allowing`undefined` as a value

Reply:
you broke this https://github.com/preactjs/preact/issues/1953

Reply:
There is no change that happened here to ref, look at https://github.com/preactjs/preact/pull/1955/files please

Reply:
Yes sorry, I mixed up the tabs and diffs, you just changed it back and forth. I landed here because you referenced the issue and the type error is still there.
But the ref **?** was removed here initially: #1447 (from what I see)